docs/howtos/DataPlane Generation - DPG/09versioning.mdx

```diff
@@ -0,0 +1,539 @@
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Versioning
+
+This doc details what emitters will generate for versioned specs
+
+## Single api Version
+
+If there is just one api version in the spec, we will generate the api surface area for that one version.
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+import "@typespec/versioning";
+
+using TypeSpec.Versioning;
+
+@versioned(My.Service.Versions)
+@service
+namespace My.Service;
+
+enum Versions {
+  v2023_11_01: "2023-11-01",
+}
+
+model StableModel {
+  stableFeature: string;
+}
+
+op stableFunctionality(...StableModel): void;
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```python
+import pytest
+from my.service import MyServiceClient, models
+
+client = MyServiceClient(endpoint=..., credential=...)
+# client's api_version will be "2023-11-01"
+
+stable_model = models.StableModel(stable_feature="present")
+print(stable_model)
+client.stable_functionality(stable_model) # call goes through
+
+preview_client = MyServiceClient(endpoint=..., credential=..., api_version="2023-11-01")
+# python allows you to override the api version, even if only one version is defined in the spec
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+// Java will generate a ServiceVersion enum containing the single api-version.
+public enum ServiceServiceVersion implements ServiceVersion {
+    /**
+     * Enum value 2023-11-01.
+     */
+    V2023_11_01("2023-11-01");
+
+    private final String version;
+
+    ServiceServiceVersion(String version) {
+        this.version = version;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getVersion() {
+        return this.version;
+    }
+
+    /**
+     * Gets the latest service version supported by this client library.
+     * 
+     * @return The latest {@link ServiceServiceVersion}.
+     */
+    public static ServiceServiceVersion getLatest() {
+        return V2023_11_01;
+    }
+}
```

Comment:
Could we put the public APIs (without implementation code) in doc?

The only code that matters would be the `getLatest()` API.

Reply:
updated